src/java/picard/vcf/MendelianViolationDetector.java

```diff
@@ -0,0 +1,338 @@
+            if (!haploid && (momGt.isNoCall() || momGt.getGQ() < MIN_GQ || dadGt.isNoCall() || dadGt.getGQ() < MIN_GQ)) continue;
+            if (kidGt.isNoCall()) continue;
+            if (momGt.isHomRef() && dadGt.isHomRef() && !kidGt.isHomRef()) {
+                if (kidGt.getPL()[0] < MIN_GQ) continue;
+            } else if (kidGt.getGQ() < MIN_GQ) continue;
+
+            // Also filter on the DP for each of the samples - it's possible to miss hets when DP is too low
+            if (haploid && ( kidGt.getDP() < MIN_DP || haploidParentalGenotype.getDP()<MIN_DP)) continue;
+            if (!haploid && ( kidGt.getDP() < MIN_DP || momGt.getDP() < MIN_DP || dadGt.getDP() < MIN_DP)) continue;
+
+            trio.NUM_VARIANT_SITES++;
+
+            ///////////////////////////////////////////////////////////////
+            // First test for haploid violations
+            ///////////////////////////////////////////////////////////////
+            MendelianViolation type = null;
+            if (haploid) {
+                if (kidGt.isHet()) continue; // Should not see heterozygous calls at haploid regions
+
+                if (!haploidParentalGenotype.getAlleles().contains(kidGt.getAllele(0))) {
+                    if (kidGt.isHomRef()) {
+                        type = MendelianViolation.Haploid_Other;
+                        trio.NUM_HAPLOID_OTHER++;
+                    } else {
+                        type = MendelianViolation.Haploid_Denovo;
+                        trio.NUM_HAPLOID_DENOVO++;
+                    }
+                }
+            }
+            ///////////////////////////////////////////////////////////////
+            // Then test for diploid mendelian violations
+            ///////////////////////////////////////////////////////////////
+            else if (isMendelianViolation(momGt, dadGt, kidGt)) {
+                if (momGt.isHomRef() && dadGt.isHomRef() && !kidGt.isHomRef()) {
+                    trio.NUM_DIPLOID_DENOVO++;
+                    type = MendelianViolation.Diploid_Denovo;
+                } else if (momGt.isHomVar() && dadGt.isHomVar() && kidGt.isHet()) {
+                    trio.NUM_HOMVAR_HOMVAR_HET++;
+                    type = MendelianViolation.HomVar_HomVar_Het;
+                } else if (kidGt.isHom() && ((momGt.isHomRef() && dadGt.isHomVar()) || (momGt.isHomVar() && dadGt.isHomRef()))) {
+                    trio.NUM_HOMREF_HOMVAR_HOM++;
+                    type = MendelianViolation.HomRef_HomVar_Hom;
+                } else if (kidGt.isHom() && ((momGt.isHom() && dadGt.isHet()) || (momGt.isHet() && dadGt.isHom()))) {
+                    trio.NUM_HOM_HET_HOM++;
+                    type = MendelianViolation.Hom_Het_Hom;
+                } else {
+                    trio.NUM_OTHER++;
+                    type = MendelianViolation.Other;
+                }
+            }
+
+            // Output a record into the family's violation VCF
```

Comment:
yes...not sure how to get around it...these could be written to disk and then merged...but that would be much work.